numpy/core/tests/test_cython.py

```diff
@@ -162,3 +162,49 @@ def test_get_datetime_iso_8601_strlen(self, install_temp):
         # uses NPY_FR_ns
         res = checks.get_datetime_iso_8601_strlen()
         assert res == 48
+
+
+def test_npyiter_attributes(install_temp):
```

Comment:
Not really attributes?  I might prefer smaller tests in general.  OTOH, this compiles, so maybe it is better to do a larger here for once.

Reply:
There's not really much to test individually so I figured a single function is better. I renamed it to `test_npyiter_api` in https://github.com/numpy/numpy/pull/24691/commits/b3430c8fd9dfbbd42c24f68f5e5906e53963cafb#diff-77fb17c954b6d12dab5aa5f47604b36c995641c8d1aed127680565a48a99e8f0